bin/portal-bridge/src/bridge/e2hs.rs

```diff
@@ -0,0 +1,439 @@
+        for block_tuple in block_stream {
+            let block_number = block_tuple
+                .header_with_proof
+                .header_with_proof
+                .header
+                .number;
+            if let Some((start, end)) = block_range {
+                if block_number < start || block_number > end {
+                    debug!("Skipping block {block_number} outside of range");
+                    continue;
+                }
+            }
+            if let Err(err) = self.validate_block_tuple(&block_tuple) {
+                error!("Failed to validate block tuple: {err:?}");
+                continue;
+            }
+            let permit = self
+                .gossip_semaphore
+                .clone()
+                .acquire_owned()
+                .await
+                .expect("Failed to acquire gossip semaphore");
+            self.metrics.report_current_block(block_number as i64);
+            let handle = Self::spawn_serve_block_tuple(
+                self.portal_client.clone(),
+                block_tuple,
+                permit,
+                self.metrics.clone(),
+            );
+            serve_block_tuple_handles.push(handle);
+        }
+        join_all(serve_block_tuple_handles).await;
+    }
+
+    fn spawn_serve_block_tuple(
+        portal_client: HttpClient,
+        block_tuple: BlockTuple,
+        permit: OwnedSemaphorePermit,
+        metrics: BridgeMetricsReporter,
+    ) -> JoinHandle<()> {
+        let number = block_tuple
+            .header_with_proof
+            .header_with_proof
+            .header
+            .number;
+        info!("Spawning serve block tuple for block {number}");
+        let block_stats = Arc::new(Mutex::new(HistoryBlockStats::new(number)));
+        tokio::spawn(async move {
+            let timer = metrics.start_process_timer("spawn_serve_block_tuple");
+            match timeout(
+                SERVE_BLOCK_TIMEOUT,
+                Self::serve_block_tuple(
+                    portal_client,
+                    block_tuple,
+                    block_stats.clone(),
+                    metrics.clone(),
+                ),
+            )
+            .await
+            {
+                Ok(Ok(())) => {
+                    info!("Served block {number}");
+                }
+                Ok(Err(e)) => {
+                    error!("Failed to serve block {number}: {e:?}");
+                }
+                Err(_) => {
+                    error!("Timed out serving block {number}");
+                }
+            }
+            metrics.stop_process_timer(timer);
+            drop(permit);
+        })
+    }
+
+    async fn serve_block_tuple(
+        portal_client: HttpClient,
+        block_tuple: BlockTuple,
+        block_stats: Arc<Mutex<HistoryBlockStats>>,
+        metrics: BridgeMetricsReporter,
+    ) -> anyhow::Result<()> {
+        let header_number = block_tuple
```

Comment:
```suggestion
        let block_number = block_tuple
```
I think this is more clear